### PR TITLE
feat: added push notifications using firebase cloud messaging

### PR DESCRIPTION
**Pull request by ibtesam-onefm**
### Requirements
As a user I want to push notifications on my mobile web so that timely information can be delivered

### Changes
- Configured push notifications using firebase
- Stored users fcm token
- Added service worker and handled background notifications

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,10 +1,110 @@
 <script setup>
-import { RouterLink, RouterView } from 'vue-router'
```
Comment (NoBoneZ) [inline src/App.vue:2]: Please, why did you remove this

Comment (ibtesam-onefm) [reply src/App.vue:2]: "RouterLink" was unused thats why removed it and for "RouterView" you can have a look at line 3 of this file, its imported over there.

**Review by NoBoneZ — APPROVED** [review]
